### ISSUE TITLE
create_network_interface() should use default VPC Security Group if no groups passed

### ISSUE THREAD

**Issue by seansummers** [spulec/moto#1807]
## create_network_interface() creates an invalid ENI -- missing a Security Group

ENI objects are required to have at least one Security Group, and `Groups` is not a required parameter. If `create_network_interface` is called with no `Groups` parameter, the VPC Default Security Group is supposed to be automatically added.

This behavior creates invalid ENI objects (with no Security Group attached), and does not match AWS EC2 behavior.

  - This code behaves differently when run with the `moto` package available or not (Python 3):
```
ec2 = boto3.resource('ec2')
vpc = tuple(ec2.vpcs.all())[0]
vpc_default_sg = next(_.id for _ in vpc.security_groups.all() if _.group_name == 'default')
print(f"VPC ({vpc.id}) default Security Group: {vpc_default_sg}")
eni = tuple(vpc.subnets.all())[0].create_network_interface()
eni_all_sgs = tuple(_['GroupId'] for _ in eni.groups)
print(f"ENI ({eni.id}) Security Group: {eni_all_sgs}")
eni.delete()
```
   - **what you expected to happen**: identical behavior for AWS EC2 and mock_ec2
   - **what actually happens**: create_network_interface creates ENIs with no security groups attached (impossible ENI irl)
   - python 3.6.6, moto 1.3.5

**Comment by spulec** [comment]
Thanks; I've changed the title and set this as a feature request for us to add.

**Comment by bblommers** [comment]
This was implemented with #4307, and part of Moto >= 2.2.14.dev18

#4720 contains a specific unit test for this scenario to verify that it now works.
